### DISCUSSION TITLE
Parameters are uppercased when rendering doc-strings with Numpydoc

### DISCUSSION THREAD

**Discussion by john-hen** [pradyunsg/furo#296] [Potential Issue] [answered]
When using the Sphinx extension Numpydoc alongside Autodoc to render NumPy-style doc-strings, parameter names and their types appear in uppercase on the page, regardless of how they are defined in the code.

To reproduce, create this `conf.py`
```python
extensions = ['sphinx.ext.autodoc', 'numpydoc']
html_theme = 'furo'
import sys
sys.path.insert(0, '.')
```
this `index.rst`
```rst
.. automodule:: module
    :members:
```
and this `module.py`
```python
def function(index, text):
    """
    Doc-string with parameter list in NumPy style.

    Parameters
    ----------
    index: int
        An index number.
        
    text : str
        Some text.
    """
```

Then run `sphinx-build . output` in the same folder. The output is this:

![screenshot](https://user-images.githubusercontent.com/65193098/140819842-a3e40c50-b08e-4fb9-a50d-67b51fecf26b.png)

The parameter names and types are in all-caps, which is probably not intended, given how confusing that might be in API documentation.

The reason seems to be that the CSS selector `dl.py .field-list dt` defined by Furo in [`_api.sass`] applies `text-transform: uppercase` not just to the "Parameters" heading, but also to the list items rendered by [Numpydoc] as the following HTML:
```html
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>index: int</strong></dt>
<dd><p>An index number.</p></dd>
<dt><strong>text</strong><span class="classifier">str</span></dt>
<dd><p>Some text.</p></dd>
</dl></dd></dl>
```

Instead of Numpydoc, we can also render Numpy-style doc-strings with [Napoleon], simply by replacing `'numpydoc'` with `'sphinx.ext.napoleon'` in `conf.py`. Then parameter names are properly cased as Napoleon produces this HTML:
```html
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>index</strong> (<em>int</em>) – An index number.</p></li>
<li><p><strong>text</strong> (<em>str</em>) – Some text.</p></li>
</ul></dd></dl>
```

Tested with Furo 2021.10.9, Numpydoc 1.1.0, Sphinx 4.2.0.

See also:
* Earlier discussion here, #200, missing a reproducible example.
* [Stack Overflow question][question] that prompted this report.

[`_api.sass`]: https://github.com/pradyunsg/furo/blob/fcd6ff66f45f6fb66d26428fe01e9b4a216ffed2/src/furo/assets/styles/content/_api.sass
[Numpydoc]:    https://numpydoc.readthedocs.io
[Napoleon]:    https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html
[question]:    https://stackoverflow.com/questions/69873561

**Comment by pradyunsg** [comment, marked as answer]
Fixed in https://github.com/pradyunsg/furo/commit/20a83d154a206f2d7d09e25337b950ce49a84084, I think.

Can you `pip install https://github.com/pradyunsg/furo/archive/refs/heads/main.zip` and confirm whether that fix works?

**Reply by john-hen** [reply]
Full disclosure, I can't seem to get Furo to build from source on either Windows or Ubuntu. (And for different reasons according to the error messages.) So I just edited the `>` child selector into the minimized `furo.css` in Furo 2021.11.23 as installed from PyPI.

But then it works. 👍 Everything looking good with Numpydoc, Napoleon, and field lists.

**Reply by pradyunsg** [reply]
If you could share the errors in a new discussion topic, that would be great!
